Source/FieldSolver/ImplicitSolvers/ThetaImplicitEM.cpp

```diff
@@ -102,69 +102,67 @@ void ThetaImplicitEM::OneStep ( const amrex::Real  a_time,
         }
     }
 
-    const amrex::Real theta_time = a_time + m_theta*m_dt;
-
     // Solve nonlinear system for Eg at t_{n+theta}
     // Particles will be advanced to t_{n+1/2}
     m_E.Copy(m_Eold); // initial guess for Eg^{n+theta}
-    m_nlsolver->Solve( m_E, m_Eold, theta_time, m_theta*m_dt );
+    m_nlsolver->Solve( m_E, m_Eold, start_time, m_theta*m_dt );
 
     // Update WarpX owned Efield_fp and Bfield_fp to t_{n+theta}
-    UpdateWarpXFields( m_E, theta_time );
+    UpdateWarpXFields( m_E, start_time );
 
     // Advance particles from time n+1/2 to time n+1
     m_WarpX->FinishImplicitParticleUpdate();
 
     // Advance Eg and Bg from time n+theta to time n+1
-    const amrex::Real new_time = a_time + m_dt;
-    FinishFieldUpdate( new_time );
+    const amrex::Real end_time = start_time + m_dt;
+    FinishFieldUpdate( end_time );
 
 }
 
 void ThetaImplicitEM::ComputeRHS ( WarpXSolverVec&  a_RHS,
                              const WarpXSolverVec&  a_E,
-                                   amrex::Real      a_time,
+                                   amrex::Real      start_time,
                                    int              a_nl_iter,
                                    bool             a_from_jacobian )
 {
     // Update WarpX-owned Efield_fp and Bfield_fp using current state of
     // Eg from the nonlinear solver at time n+theta
-    UpdateWarpXFields( a_E, a_time );
+    UpdateWarpXFields( a_E, start_time );
 
     // Update particle positions and velocities using the current state
     // of Eg and Bg. Deposit current density at time n+1/2
-    m_WarpX->ImplicitPreRHSOp( a_time, m_dt, a_nl_iter, a_from_jacobian );
+    const amrex::Real theta_time = start_time + m_theta*m_dt;
+    m_WarpX->ImplicitPreRHSOp( theta_time, m_dt, a_nl_iter, a_from_jacobian );
 
     // RHS = cvac^2*m_theta*dt*( curl(Bg^{n+theta}) - mu0*Jg^{n+1/2} )
     m_WarpX->ImplicitComputeRHSE( m_theta*m_dt, a_RHS);
 }
 
 void ThetaImplicitEM::UpdateWarpXFields ( const WarpXSolverVec&  a_E,
-                                          amrex::Real            a_time )
+                                          amrex::Real start_time )
 {
-    amrex::ignore_unused(a_time);
 
     // Update Efield_fp owned by WarpX
-    m_WarpX->SetElectricFieldAndApplyBCs( a_E );
+    const amrex::Real theta_time = start_time + m_theta*m_dt;
+    m_WarpX->SetElectricFieldAndApplyBCs( a_E, theta_time );
 
     // Update Bfield_fp owned by WarpX
     ablastr::fields::MultiLevelVectorField const& B_old = m_WarpX->m_fields.get_mr_levels_alldirs(FieldType::B_old, 0);
-    m_WarpX->UpdateMagneticFieldAndApplyBCs( B_old, m_theta*m_dt );
+    m_WarpX->UpdateMagneticFieldAndApplyBCs( B_old, m_theta*m_dt, start_time );
 
 }
 
-void ThetaImplicitEM::FinishFieldUpdate ( amrex::Real  a_new_time )
+void ThetaImplicitEM::FinishFieldUpdate ( amrex::Real end_time )
```

Comment:
Here too, clang-tidy complains about the fact that we use `end_time` in the definition here and `a_new_time` in the declaration in ThetaImplicitEM.H.

Reply:
Fixed now. Too many changes to keep track of and the local compilation doesn't check this.